File: _scripts/userscript-devtoprod.pl
#!/usr/bin/perl -w

##### MODULES  #####{
use File::Basename;
use Cwd qw(abs_path);

use File::Slurp;

use feature 'say';

use strict;
use warnings;
# use diagnostics;
####################}

if($> != 0) { die("Script must be run as root!"); }

my $trackrLocation = dirname(abs_path(__FILE__)) =~ s/[\\|\/]_scripts$//r;

my $userscriptDev  = read_file($trackrLocation."/public/userscripts/manga-tracker.dev.user.js");
my $userscriptProd = ($userscriptDev =~ s/manga-tracker\.localhost:20180/trackr\.moe/rg =~ s/http:\/\/trackr\.moe/https:\/\/trackr.moe/rg);

my $goodReplace  = "=== 'http://manga-tracker.localhost:20180'";
my $badReplace = "=== 'https://trackr.moe'";
$userscriptProd =~ s/$badReplace/$goodReplace/g;

overwrite_file($trackrLocation."/public/userscripts/manga-tracker.user.js", {binmode => ':utf8'}, $userscriptProd);

File: _scripts/userscript_devtoprod.php
<?php

(PHP_SAPI !== 'cli' || isset($_SERVER['HTTP_USER_AGENT'])) && die('CLI only.');

chdir(dirname(__FILE__, 2)); //Just to make things easier, change dir to project root.

$userscriptDev = file_get_contents('./public/userscripts/manga-tracker.dev.user.js');

// Replace where needed.
$userscriptDev = str_replace('http://manga-tracker.localhost:20180/userscripts', 'https://trackr.moe/userscripts', $userscriptDev);

// Push to prod file.
file_put_contents('./public/userscripts/manga-tracker.user.js', $userscriptDev);
